aips/aip-x.md

```diff
@@ -0,0 +1,100 @@
+---
+aip: 
+title: Private Entry Function for Multisig Account Creation
+author: yeptos (https://github.com/yeptos), gregnazario, lightmark?
+discussions-to (*optional): <a url pointing to the official discussion thread>
+Status: Draft
+last-call-end-date (*optional): <mm/dd/yyyy the last date to leave feedbacks and reviews>
+type: Standard (Framework)
+created: 9/10/2024
+updated (*optional): <mm/dd/yyyy>
+requires (*optional): <AIP number(s)>
+---
+
+# AIP-X - Private Entry Function for Multisig Account Creation
+
+## Summary
+
+This AIP proposes adding new private entry functions to allow an existing Aptos account to create a multisig account using a signer, without requiring complex steps to generate a signature of a struct as proof. Currently, only accounts with Ed25519 and MultiEd25519 authentication schemes can migrate to multisig accounts. This enhancement will enable any account with any authentication key scheme to create a multisig account on top of it, providing a straightforward migration path from any accounts to multisig v2 accounts.
+
+### Out of Scope
+
+This AIP does not cover changes to other aspects of multisig account functionality beyond the creation process from an existing account.
+
+## High-level Overview
+
+The proposal introduces two new private entry functions in the `multisig_account` module:
+
+- `create_with_existing_account_call`: Creates a new multisig account on top of an existing account without revoking the original auth key.
+
+- `create_with_existing_account_and_revoke_auth_key_call`: Creates a new multisig account on top of an existing account and immediately rotates the original auth key to 0x0.
+
+These functions simplify the process of creating a multisig account from an existing account, regardless of its authentication key scheme. This is particularly useful for migrating from multi-key accounts to multisig accounts, which is currently not possible.
```

Comment:
I would mention this mirrors the changes here: https://github.com/aptos-labs/aptos-core/commit/d771cec3c4638a9a5d50120765f921e47feea83f